### ISSUE TITLE
Devices do not support shot batching

### ISSUE THREAD

**Issue by trbromley** [PennyLaneAI/pennylane-sf#74]
#### Issue description

https://github.com/PennyLaneAI/pennylane/pull/1103 was merged a few months ago to support shot batching. This involves processing the input shots into a total number of shots (specified by `self._shots`) and a shots vector (specified by `self._shot_vector`).

Currently the PennyLane-SF devices all inherit from `StrawberryFieldsSimulator`. In its `expval` method, there is no usage of the `self._shot_vector` attribute - the method just samples based on the total number of shots and returns the average.

Consider

```python
import pennylane as qml

dev = qml.device("strawberryfields.fock", wires=1, cutoff_dim=10, shots=[50, 50])

@qml.qnode(dev)
def qnode(x):
    # Encode input x into quantum state
    qml.Displacement(x, 0.0, wires=0)

    return qml.expval(qml.X(0))
```
The expected return is
```pycon
>>> qnode(0.2)
tensor([0.36074875, 0.40190935], requires_grad=True)
```
That is, two repetitions with 50 shots. The actual output is
```pycon
>>> qnode(0.2)
tensor(0.53289407, requires_grad=True)
```
That is, one repetition with 100 shots.

**Comment by orielkiss** [comment]
Hi,
I just wanted to know if shot batching provides any computational advantage compared to a for loop on the user's side? Thanks

**Comment by co9olguy** [comment]
Hi @orielkiss, there's no specific computational advantage, no. But shot batching does allow one to "do more with less" i.e., to get the outputs from several possible experiments in one single submission to an online quantum device

**Comment by mariaschuld** [comment]
Getting back to this issue: how hard would it be to implement?

@co9olguy to clarify your answer further: As far as I know, shot batching gives you a huge speedup on simulators (not only remote communication overhead), because the final state is only simulated once, but lots of shots are sampled. If you're after samples in the first place, this could be catered for by implementing `sample` measurements on CV devices. However, the original use case for shot batching is when users want a 1-shot estimation of an expectation value. In principle, this should be the same as a sample, but it has a clean definition of a gradient and is hence trainable! 

In short, implementing `sample` measurements and shot batching on CV devices will result in a massive speedup compared to loops over circuit evaluations. Maybe we should look into supporting this?

**Comment by antalszava** [comment]
Devices that inherit from `Device` (instead of `QubitDevice`) do not support shot batching. The devices in this plugin inherit from `Device`. Due to a recent addition, a [`UserWarning` is being raised in Pennylane](https://github.com/PennyLaneAI/pennylane/pull/1659) each time shot batching is being used for such devices.

See [the comment from the original PR adding shot batching support](https://github.com/PennyLaneAI/pennylane/pull/1103#issuecomment-784287830) and the original PR: https://github.com/PennyLaneAI/pennylane/pull/1103.

It would make sense to add this functionality to PennyLane core. Opened an [issue there](https://github.com/PennyLaneAI/pennylane/issues/1696) and closing this one.
